### PR TITLE
full review post-trial 01 and new episodes in tutorials-early

### PR DESCRIPTION
**Pull request by avallecam**
This PR aims to collect edit suggestions for the tutorials-early repository.

This includes any specific addition to:
- new episodes added on read, clean, validate, and aggregate data
- add content on epidemiological background or field applications
- add conceptual explanations of methods

All rendered content is in https://epiverse-trace.github.io/tutorials-early/

### CODE REVIEW

File: episodes/describe-cases.Rmd [hunk 1]
```diff
@@ -0,0 +1,221 @@
+---
+title: 'Aggregate and visulaize'
```
Comment (Bisaloo) [inline episodes/describe-cases.Rmd:2]:
```suggestion
title: 'Aggregate and visuliaze'
```

**Review by Degoot-AM — CHANGES_REQUESTED** [review]
@avallecam, I have a few comments on this PR, please review the changes.

File: .github/workflows/README.md [hunk 2]
```diff
@@ -96,7 +96,7 @@ are okay.
 
 This update is run ~~weekly or~~ on demand.
 
-### 03 Maintain: Update Pacakge Cache (update-cache.yaml)
+### 03 Maintain: Update Package Cache (update-cache.yaml)
 
 For lessons that have generated content, we use {renv} to ensure that the output
 is stable. This is controlled by a single lockfile which documents the packages
```
[the remark this answers is not in the document]
Comment (avallecam) [reply .github/workflows/README.md:102]: interesting, this looks fixed in the [online file](https://github.com/epiverse-trace/tutorials-early/blob/main/.github/workflows/README.md#03-maintain-update-package-cache-update-cacheyaml)

File: config.yaml [hunk 3]
```diff
@@ -11,29 +11,29 @@
 carpentry: 'incubator'
 
 # Overall title for pages.
-title: 'Lesson Title' # FIXME
+title: 'Accessing and using delays to Quantify transmission'
```
[the remark this answers is not in the document]
Comment (avallecam) [reply config.yaml:14]: seems to be fixed in https://github.com/epiverse-trace/tutorials-early/commit/d0d3b0857d82a588794c79375a84a3a0299e26cb

File: episodes/read-cases.Rmd [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+exercises: 10
+editor_options: 
+  chunk_output_type: inline
+---
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- Where do you usually store your outbreak data?
+- How many different data formats can I read? 
+- Is it possible to import data from database and health APIs?
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:13]: do you suggest this edit?
```suggestion
- Is it possible to import data from databases and health APIs? 
```

File: episodes/read-cases.Rmd [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+
+```{r,eval=TRUE, echo=FALSE, message=FALSE}
+# read data
+case_data <- rio::import(file.path("data", "ebola_cases.csv"))
+# preview data
+head(case_data, 5)
+```
+
+Similarly, you can import files of other formats such as `tsv`, `xlsx`, etc.
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:61]: may the original one be clearer?

File: episodes/read-cases.Rmd [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+::::::::::::::::::::::::::::::::: challenge
+
+###  Reading compressed data 
+
+Take 1 minute:
+- Is it possible to read compressed data in `R`?
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:68]:
```suggestion

- Can you read data from a compressed file in `R`?
```

[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:68]: I suggest adding the text to download the file in this same line too.

```suggestion
Can you read data from a compressed file in `R`? Download this [zip file](https://epiverse-trace.github.io/tutorials-early/data/Marburg.zip) containing data for Marburg outbreak and then import it to your working environment.
```

File: episodes/read-cases.Rmd [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+::::::::::::::::::::::
+
+::::::::::::::::: solution
+
+```{r,eval=FALSE}
+rio::import(here::here("some", "where", "downto", "path", "file_name.zip"))
+```
+Click [here](https://github.com/epiverse-trace/tutorials-early/tree/md-outputs-PR-39/data/Marburg.zip) to download a zip file containing data for Marburg outbreak  and then import it to your working environment.
```
Comment (avallecam) [inline episodes/read-cases.Rmd:87]:
```suggestion
Click [here](https://epiverse-trace.github.io/tutorials-early/data/Marburg.zip) to download a zip file containing data for Marburg outbreak  and then import it to your working environment.
```

File: learners/setup.md [hunk 8]
```diff
@@ -2,53 +2,228 @@
...
+### Download the data
 
-:::::::::::::::: solution
+We will download the data directly from R during the tutorial. However, if you are expecting problems with the network, it may be better to download the data beforehand and store it on your machine.
 
-### Linux
+The data files for the tutorial can be downloaded manually here: 
 
-Use Terminal
+- <https://epiverse-trace.github.io/tutorials/data/ebola_cases.csv>
```
Comment (avallecam) [inline learners/setup.md:225]:
```suggestion
- <https://epiverse-trace.github.io/tutorials-early/data/ebola_cases.csv>
- <https://epiverse-trace.github.io/tutorials-early/data/Marburg.zip>
```

File: .github/workflows/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (avallecam) [reply .github/workflows/README.md:102]: > This is controlled by a single lockfile, which documents versions of the packages
> needed for the lessons.

can you rephrase if there is an issue you identify with this? It is the workflow that can generate PR to update package versions in the [lockfile](https://github.com/epiverse-trace/tutorials-early/blob/main/renv/profiles/lesson-requirements/renv.lock)

File: episodes/clean-data.Rmd [hunk 9]
```diff
@@ -0,0 +1,299 @@
+---
+title: 'Clean and validate'
+teaching: 20
+exercises: 10
+---
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- How to clean and standardize case data?
+- How to convert raw dataset into a `linelist` object?
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/clean-data.Rmd:10]: is there any specific change proposed here?

File: episodes/clean-data.Rmd [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+- What differences you can observe in the column names?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+If you want to maintain certain column names without subjecting them to the standardization process, you can utilize the `keep` parameter of the `standardize_column_names()` function. This parameter accepts a vector of column names that are intended to be kept unchanged.
+
+**Exercise:** Standardize the column names of the input dataset, but keep the “V1” column as it is.
```
Comment (avallecam) [inline episodes/clean-data.Rmd:86]: Could this also be within a callout box as challenge?

```suggestion
::::::::::::::::::::::::::::::::::::: challenge

Standardize the column names of the input dataset, but keep the “V1” column as it is.

::::::::::::::::::::::::::::::::::::::::::::::::
```

File: learners/setup.md [hunk 11]
```diff
@@ -2,53 +2,229 @@
...
+During the tutorial, we will need a number of R packages. Packages contain useful R code written by other people. We will use packages from the [Epiverse-TRACE](https://epiverse-trace.github.io/). 
 -->
-Download the [data zip file](https://example.com/FIXME) and unzip it to your Desktop
 
-## Software Setup
+Open RStudio and **copy and paste** the following code chunk into the [console window](https://docs.posit.co/ide/user/ide/guide/code/console.html), then press the <kbd>Enter</kbd> (Windows and Linux) or <kbd>Return</kbd> (MacOS) to execute the command:
+
+```r
+if(!require("pak")) install.packages("pak")
+
```
Comment (avallecam) [inline learners/setup.md:107]:
```suggestion
# for episodes on read, clean, validate and visualize linelist
new_packages <- c(
  "epiverse-trace/cleanepi",
  "rio",
  "here",
  "DBI",
  "RSQLite",
  "dbplyr",
  "linelist",
  "epiverse-trace/simulist",
  "incidence2",
  "epiverse-trace/tracetheme",
  "tidyverse"
)

pak::pkg_install(new_packages)

# for episodes on access delays and quantify transmission
```

**Review by avallecam — COMMENTED** [review]
@Degoot-AM I finally managed to go through all the first four episodes! thank you work on them!

I prioritized code flow, that everything is running with the recent package versions. I found some edits required for cleanepi code, and one that I reported as issue.

Then I suggested to add some space between code lines within long chunks.

I added namespace reminders and some callout box of prerequisites as reminders of the dataset that we need to download before going through the episode.

For the first episode on read case data, we may open it to four subtitles to add brief details on each step. Also I suggested a figure as a first intro to databases.

Regarding the setup page, better if we focus to update the download links and the packages required for the four episodes. We had mixed and overlapping suggestions here and in the PR #64 so better if we reconciliate them after merging these most recent edits.

Are you able to approve commits in this PR? If yes, please, go ahead!

File: episodes/read-cases.Rmd [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+:::::::::::::::::::::::::::::::::::::::::::
+
+
+## Reading from databases
+
+The [DBI](https://dbi.r-dbi.org/) package serves as a versatile interface for interacting with database management 
+systems (DBMS) across different back-ends or servers. It offers a uniform method for accessing and retrieving data from various database systems.
+
+
+The following code chunk demonstrates how to create a temporary SQLite database in memory, store the `case_data` as a table on it, and subsequently read it:
```
Comment (avallecam) [inline episodes/read-cases.Rmd:99]:
```suggestion
The following code chunk demonstrates how to create a temporary SQLite database in memory, store the `ebola_confirmed` as a table on it, and subsequently read it:
```

File: episodes/read-cases.Rmd [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+
+The below example shows how to import a `csv` file into `R` environment using `{rio}` package.
+
+```{r,eval=FALSE,echo=TRUE}
+library("rio")
+library("here")
+
+# read data
+# e.g., the path to our file is data/raw-data/ebola_cases.csv then:
+ebola_confirmed <- read_csv(here::here("data", "raw-data", "ebola_cases.csv"))
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:47]: this edit is related to https://github.com/epiverse-trace/tutorials-early/issues/73. I found this facilitates running the code.

File: episodes/read-cases.Rmd [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::::::
+
+
+## Reading from databases
+
+The [DBI](https://dbi.r-dbi.org/) package serves as a versatile interface for interacting with database management 
+systems (DBMS) across different back-ends or servers. It offers a uniform method for accessing and retrieving data from various database systems.
+
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:97]: this is the source of figure https://jcsites.juniata.edu/faculty/rhodes/dbms/relnmodel.htm

**Review by Degoot-AM — COMMENTED** [review]
Thank you @avallecam for you feedback.

**Review by Degoot-AM — APPROVED** [review]

File: episodes/clean-data.Rmd [hunk 15]
```diff
@@ -0,0 +1,299 @@
...
+```
+
+Note that, our simulated Ebola does not contain duplicated nor constant rows or columns. 
+
+### Replacing missing values
+
+In addition to the regularities, raw data can contain missing values that may be encoded by different strings, including the empty. To ensure robust analysis, it is a good practice to replace all missing values by `NA` in the entire dataset. Below is a code snippet demonstrating how you can achieve this in `{cleanepi}`:
+
+```{r}
+sim_ebola_data <- cleanepi::replace_missing_values(sim_ebola_data)
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/clean-data.Rmd:125]:
```suggestion
sim_ebola_data <- cleanepi::replace_missing_values(sim_ebola_data, na_strings = "")
```

File: .github/workflows/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (avallecam) [reply .github/workflows/README.md:102]: since this file is a workflow file from the workbench infrastructure, better to propose edits in the [carpentries template file](https://github.com/carpentries/workbench-template-rmd/tree/main/.github/workflows)

File: episodes/clean-data.Rmd [hunk 16]
```diff
@@ -0,0 +1,320 @@
...
+
+- How to clean and standardize case data?
+- How to convert raw dataset into a `linelist` object?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Explain how to clean, curate, and standardize case data using `{cleanepi}` package
+- Demonstrate how to covert case data to `linelist` data
```
Comment (avallecam) [inline episodes/clean-data.Rmd:17]:
```suggestion
- Demonstrate how to covert case data to `linelist` data 
- Perform essential data-cleaning operations to address epidemiologically-related inconsistencies.
```

File: episodes/clean-data.Rmd [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+- How to clean and standardize case data?
+- How to convert raw dataset into a `linelist` object?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Explain how to clean, curate, and standardize case data using `{cleanepi}` package
+- Demonstrate how to covert case data to `linelist` data 
+
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/clean-data.Rmd:18]: added an edit suggestion of this in https://github.com/epiverse-trace/tutorials-early/pull/70/files#r1640396788

File: learners/setup.md [hunk 18]
```diff
@@ -2,53 +2,229 @@
...
+The tutorials are built around an outbreak analysis pipeline split into three stages: **Early tasks**, **Middle tasks** and **Late tasks**.
+
+![An overview of the tutorial topics](https://epiverse-trace.github.io/task_pipeline-minimal.svg)
+
+Each task has its tutorial website. Each tutorial website consists of a set of episodes.
+
+| [Early task tutorials ➠](https://epiverse-trace.github.io/tutorials-early/) | [Middle task tutorials ➠](https://epiverse-trace.github.io/tutorials-middle) | [Late task tutorials ➠](https://epiverse-trace.github.io/tutorials-late/) |
+|---|---|---|
+| Read and clean case data, and make linelist | Real-time analysis and forecasting | Scenario modelling |
+| Convert linelist data to incidence, access delay distributions, and estimate transmission metrics. | Forecast cases, estimate severity and  superspreading. | Simulate disease spread and investigate interventions. |
```
Comment (avallecam) [inline learners/setup.md:22]:
```suggestion
| Read, clean and validate case data, convert linelist data to incidence for visualization. | Access delay distributions and estimate transmission metrics, forecast cases, estimate severity and superspreading. | Simulate disease spread and investigate interventions. |
```

File: episodes/read-cases.Rmd [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+```{r,eval=FALSE}
+rio::import(here::here("some", "where", "downto", "path", "file_name.zip"))
+```
+Click [here](https://epiverse-trace.github.io/tutorials-early/data/Marburg.zip) to download a zip file containing data for Marburg outbreak  and then import it to your working environment.
+::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::::::
+
+
+## Reading from databases
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:119]: placed as issue in #77

File: episodes/read-cases.Rmd (hunk 14 above)
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:123]: as issue in #78

File: episodes/read-cases.Rmd [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+
+# Store the 'case_data' dataframe as a table named 'cases'
+# in the SQLite database
+DBI::dbWriteTable(db_con, "cases", case_data)
+# Read data from the 'cases' table
+result <- DBI::dbReadTable(db_con, "cases")
+# Close the database connection
+DBI::dbDisconnect(db_con)
+# View the result
+base::print(utils::head(result))
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/read-cases.Rmd:155]: issue in #79

File: episodes/clean-data.Rmd [hunk 21]
```diff
@@ -0,0 +1,299 @@
...
+
+Further more, you can combine multiple data cleaning tasks via the pipe operator in "|>", as shown in the below code snippet. 
+```{r}
+# remove the line below once Karim has updated cleanepi
+raw_ebola_data$`case id` <- as.character(raw_ebola_data$`case id`)
+# PERFORM THE OPERATIONS USING THE pipe SYNTAX
+cleaned_data <- raw_ebola_data |>
+  cleanepi::standardize_column_names(keep = "V1", rename = NULL) |>
+  cleanepi::replace_missing_values(target_columns = NULL) |>
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/clean-data.Rmd:263]:
```suggestion
  cleanepi::replace_missing_values(na_strings = "") |>
```

**Comment by avallecam** [comment]
redirecting this PR to main now
